src/control/server/harness.go

```diff
@@ -223,6 +224,19 @@ func (h *IOServerHarness) Start(parent context.Context) error {
 			if err := instance.StartManagementService(); err != nil {
 				return errors.Wrap(err, "failed to start management service")
 			}
+			// registered instance with system membership
+			m, err := instance.newMember()
+			if err != nil {
+				return errors.Wrap(err, "failed to get member from instance")
+			}
+			count, err := membership.Add(m)
+			if err != nil {
+				return errors.Wrap(err, "failed to add MS replica to membership")
+			}
+			if count != 1 {
+				return errors.Errorf("expected MS replica to be first member "+
+					"(want 1, got %d)", count)
```

Comment:
There could be several MS replicas though, couldn't there? Maybe I'm misunderstanding the flow here.

Reply:
yes discussed this with Mike, currently we can't handle multiple in this situation unless we know more about the implementation of distributed MS

Reply:
Is this membership count per-node or for the entire DAOS system? Maybe that's what I'm not following. In either case, though, I'm wondering what this check is really guarding against. Is some harm done if the MS isn't the first member?

Reply:
count per system, this check is ensuring we have deterministic behaviour. this manually adds a member to the membership for the bootstrapping instance because that instance will never join (which is where membership registration happens). this check ensures it's only ever first member that registers in this way as no other member should other than the bootstrap. it's undetermined as to whether MSreplicas will join or not when we have multiple access points (but probably they will join). that any more clear?

Reply:
Got it. Thanks.